plugins/confluence.go

```diff
@@ -171,21 +176,58 @@ func (p *ConfluencePlugin) getPages(space ConfluenceSpaceResult, start int) (*Co
 	return &response.Results, nil
 }
 
-func (p *ConfluencePlugin) getContent(page ConfluencePage, space ConfluenceSpaceResult) (*Item, error) {
-	url := p.URL + "/rest/api/content/" + page.ID + "?expand=body.storage,body.view.value,version,history.previousVersion"
-	originalUrl := p.URL + "/spaces/" + space.Key + "/pages/" + page.ID
-	request, err := p.httpRequest(http.MethodGet, url)
+func (p *ConfluencePlugin) getContents(page ConfluencePage, space ConfluenceSpaceResult) (*[]Item, error) {
+	items := make([]Item, 0)
 
+	actualPage, previousVersion, err := p.getContent(page, space, 0)
 	if err != nil {
-		return nil, fmt.Errorf("unexpected error creating an http request %w", err)
+		return nil, err
+	}
+
+	items = append(items, *actualPage)
+
+	// If older versions exist & run history is true
+	for previousVersion > 0 && p.History {
+		actualPage, previousVersion, err = p.getContent(page, space, previousVersion)
+		if err != nil {
+			return nil, err
+		}
+		items = append(items, *actualPage)
+	}
+
+	return &items, nil
+}
+
+func (p *ConfluencePlugin) getContent(page ConfluencePage, space ConfluenceSpaceResult, version int) (*Item, int, error) {
+	var url string
+	var originalUrl string
+
+	// If no version given get the latest, else get the specified version
+	if version == 0 {
+		url = fmt.Sprintf("%s/rest/api/content/%s?expand=body.storage.value,version,history.previousVersion", p.URL, page.ID)
+		originalUrl = p.URL + "/spaces/" + space.Key + "/pages/" + page.ID
+
+	} else {
+		url = fmt.Sprintf("%s/rest/api/content/%s?status=historical&version=%d&expand=body.storage.value,version,history.previousVersion", p.URL, page.ID, version)
+		originalUrl = url
+	}
+
+	request, err := p.httpRequest(http.MethodGet, url)
```

Comment:
pass the query params to `httpRequest` method instead of formatting it yourself in the url

Reply:
I prefered to do like that because we are creating contents rigth after it and with that I don't need to build the entire URL twice (inside httpRequest and on content urls).